src/main/kotlin/blackjack/domain/BlackJack.kt

```diff
@@ -1,55 +1,59 @@
 package blackjack.domain
 
-class BlackJack(val players: List<Player>) {
-
-    private var nowPlayer = 0
-    private var playCount = 0
-    private var isEnd = false
+class BlackJack(
+    val players: List<Player>,
+    val dealer: Dealer = Dealer(),
+    private val gameCards: GameCards = GameCards(),
+) {
 
     fun start() {
-        for (i in 0 until START_CARD_COUNT) players.forEach { it.draw() }
+        distributeInitialCard()
     }
 
-    fun isEnd(): Boolean {
-        if (!isEnd) {
-            checkAndChangeGamePlayer()
+    private fun distributeInitialCard() {
+        for (i in 0 until START_CARD_COUNT) {
+            dealer.addCard(gameCards.draw())
+            players.forEach { it.addCard(gameCards.draw()) }
         }
-        return isEnd
+    }
+
+    fun isEnd(): Boolean {
+        return players.all { !it.canProceedTurn() }
     }
 
     fun getNowPlayer(): Player {
-        return players[nowPlayer]
-    }
-
-    fun play(answer: String): Int {
-        val count = playCount
-        when (answer) {
-            "y" -> {
-                getNowPlayer().draw()
-                playCount++
-            }
-            "n" -> changeNowPlayer()
-            else -> throw IllegalArgumentException("잘못된 답변입니다")
+        return players.first { it.canProceedTurn() }
+    }
+
+    fun playGameTurn(isPlaying: Boolean) {
+        when (isPlaying) {
+            true -> getNowPlayer().addCard(gameCards.draw())
+            false -> getNowPlayer().finishedTurn()
         }
-        return count
     }
 
-    private fun checkAndChangeGamePlayer() {
-        if (players[nowPlayer].score() > BLACKJACK_MAX_SCORE) {
-            changeNowPlayer()
+    fun askDealerForAdditionalCard(): Boolean {
+        if (dealer.score() <= DEALER_CARD_STANDARD_SCORE) {
+            dealer.addCard(gameCards.draw())
+            return true
         }
+        return false
     }
 
-    private fun changeNowPlayer() {
-        nowPlayer++
-        playCount = 0
-        if (nowPlayer >= players.size) {
-            isEnd = true
+    fun getResult(): Ranks {
+        val result = players.associateWith {
+            getPlayerRank(it)
```

Comment:
한 줄로 해결되는 코드를 굳이 메서드로 쪼갤 필요는 없을 것 같아요. 😃
지금처럼 메서드 호출 없이 로직을 바로 위치시켜도 코드가 잘 읽히는 것 같아요. 😉

```suggestion
            PlayerRank.getValues(it.score(), dealer.score())
```